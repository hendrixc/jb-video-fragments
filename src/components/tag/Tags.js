import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Chip, ChipSet } from '@rmwc/chip';
import '@material/chips/dist/mdc.chips.css';


class Tags extends Component {
  render() {
    const { tags } = this.props;
    return (
      <div className="Tag">
        <div className="group-chips">
          <ChipSet>
            {tags.map((tag, index) => (
              <Chip key={`${tag}-${index}`} icon="more" label={tag} />
            ))}
          </ChipSet>
        </div>
      </div>
    );
  }
}

Tags.propTypes = {
  tags: PropTypes.array,
};

Tags.defaultProps = {
  tags: [],
};


export default Tags;
