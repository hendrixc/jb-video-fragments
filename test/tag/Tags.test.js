import React from 'react';
import renderer from 'react-test-renderer';
import Tags from '../../src/components/tag/Tags';

it('renders an component into the Tags as snapshot', () => {
  const tagsComponent = renderer.create(<Tags />).toJSON();
  expect(tagsComponent).toMatchSnapshot();
});
