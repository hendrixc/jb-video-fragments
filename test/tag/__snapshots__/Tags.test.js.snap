// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an component into the Tags as snapshot 1`] = `
<div
  className="Tag"
>
  <div
    className="group-chips"
  >
    <div
      className="mdc-chip-set"
    />
  </div>
</div>
`;
